pkg/cloud/cloud.go

```diff
@@ -102,6 +113,23 @@ func NewCloud() (Cloud, error) {
 	}, nil
 }
 
+// NewCloudWithRole returns a new instance of AWS cloud after assuming an aws role
+// It panics if session is invalid
+func NewCloudWithRole(awsRoleArn string) (Cloud, error) {
+	sess := session.Must(session.NewSession(&aws.Config{}))
+	metadata, err := NewMetadataService(sess)
```

Comment:
So in csi-driver world, we need the metadata service to determine the region? Is there any other usage for this metadata?

Reply:
AFAIK, metadata service is being used to get region and instance Id (node Id).

Reply:
it may also be used for credentials since it's in the default chain. 

we do eventually want to make it optional though, some efforts from ebs we can copy soon. i.e., region and instance id can be determined by reading the Node object in Kubernetes API. creds can be acquired from secret or, in EKS, IAM for ServiceAccounts